policy-controller/grpc/src/lib.rs

```diff
@@ -452,3 +604,313 @@ fn convert_filter(filter: InboundFilter) -> proto::http_route::Filter {
+                (Some(_), None) => return std::cmp::Ordering::Less,
+                (None, Some(_)) => return std::cmp::Ordering::Greater,
+            };
+            by_ts.then_with(|| a_name.cmp(b_name))
+        });
+
+        let mut http_routes: Vec<_> = http_routes
+            .into_iter()
+            .map(|(name, route)| {
+                convert_outbound_http_route(
+                    outbound.namespace.clone(),
+                    name,
+                    route,
+                    backend.clone(),
+                )
+            })
+            .collect();
+
+        if http_routes.is_empty() {
+            http_routes = vec![default_outbound_http_route(backend)];
+        }
+
+        linkerd2_proxy_api::outbound::proxy_protocol::Kind::Detect(
+            outbound::proxy_protocol::Detect {
+                timeout: Some(
+                    time::Duration::from_secs(10)
+                        .try_into()
+                        .expect("failed to convert detect timeout to protobuf"),
+                ),
+                opaque: Some(outbound::proxy_protocol::Opaque {
+                    routes: Default::default(),
+                }),
+                http1: Some(outbound::proxy_protocol::Http1 {
+                    routes: http_routes.clone(),
+                }),
+                http2: Some(outbound::proxy_protocol::Http2 {
+                    routes: http_routes,
+                }),
+            },
+        )
+    };
+
+    outbound::OutboundPolicy {
+        protocol: Some(outbound::ProxyProtocol { kind: Some(kind) }),
+    }
+}
+
+fn convert_outbound_http_route(
+    namespace: String,
+    name: String,
+    OutboundHttpRoute {
+        hostnames,
+        rules,
+        creation_timestamp: _,
+    }: OutboundHttpRoute,
+    default_backend: outbound::http_route::WeightedRouteBackend,
+) -> outbound::HttpRoute {
+    let metadata = Some(Metadata {
+        kind: Some(metadata::Kind::Resource(api::meta::Resource {
+            group: "policy.linkerd.io".to_string(),
+            kind: "HTTPRoute".to_string(),
+            namespace,
+            name,
+            ..Default::default()
+        })),
+    });
+
+    let hosts = hostnames
+        .into_iter()
+        .map(http_route::convert_host_match)
+        .collect();
+
+    let rules = rules
+        .into_iter()
+        .map(|OutboundHttpRouteRule { matches, backends }| {
+            let mut backends = backends
+                .into_iter()
+                .map(convert_http_backend)
+                .collect::<Vec<_>>();
+            if backends.is_empty() {
+                backends = vec![default_backend.clone()];
+            }
+            outbound::http_route::Rule {
+                matches: matches.into_iter().map(http_route::convert_match).collect(),
+                backends: Some(outbound::http_route::Distribution {
+                    kind: Some(outbound::http_route::distribution::Kind::RandomAvailable(
```

Comment:
is `RandomAvailable` or `FirstAvailable` the preferred default (cc @olix0r)?

Reply:
We should probably use `FirstAvailable` by default. It's cheaper.